src/inference/hmc.jl

```diff
@@ -136,8 +136,8 @@ function sample_init!(
         end
     end
     
-    # Ensure h.metric has the same dim as θ.
-    spl.state.h = update(spl.state.h, spl.state.vi[spl])
```

Comment:
@cpfiffer I cannot see why the previous thing would fail. Can you please add a test that fails for this one? I'd like to debug it and see what exactly happens.

Reply:
Sure. I just added the `update` line back in, and I included a test set to check that an error is thrown. It seems to be only NUTS that's having this issue. The relevant test is here:

https://github.com/TuringLang/Turing.jl/blob/9eca4e9d7a40365247353498de3331ea013756d4/test/inference/hmc.jl#L178

For your purposes, I've made a full MWE that should work much quicker:

```julia
using Turing

@model gdemo(x, y) = begin
    s ~ InverseGamma(2, 3)
    m ~ Normal(0, sqrt(s))
    x ~ Normal(m, sqrt(s))
    y ~ Normal(m, sqrt(s))
    return s, m
  end

alg = Gibbs(PG(10, :m), NUTS(100, 0.65, :s))
sample(gdemo(1.5, 2.0), alg, 300)
```